src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -43,12 +43,19 @@ _mongoc_topology_reconcile_add_nodes (mongoc_server_description_t *sd,
                                       mongoc_topology_t *topology)
 {
    mongoc_topology_scanner_t *scanner = topology->scanner;
+   mongoc_topology_scanner_node_t *node;
+
+   /* Search by ID and update hello_ok */
+   node = mongoc_topology_scanner_get_node (scanner, sd->id);
+   if (node) {
+      node->hello_ok = sd->hello_ok;
 
-   /* quickly search by id, then check if a node for this host was retired in
-    * this scan. */
-   if (!mongoc_topology_scanner_get_node (scanner, sd->id) &&
-       !mongoc_topology_scanner_has_node_for_host (scanner, &sd->host)) {
-      mongoc_topology_scanner_add (scanner, &sd->host, sd->id);
+      return true;
+   }
+
+   /* Check if a node for this host was retired in this scan. */
+   if (!mongoc_topology_scanner_has_node_for_host (scanner, &sd->host)) {
```

Comment:
I do not fully understand the implications of changing this line, but should we change this to be equivalent to the old behavior?
```suggestion
   if (!node && !mongoc_topology_scanner_has_node_for_host (scanner, &sd->host)) {
```

Reply:
Sure thing. The previous logic was something like this:
```
if (!node_found && !node_found_for_host) {
    // Add scanner node
}

return true
```

I've updated this to:
```
if (node_found) {
    // update hello_ok
}

if (!node_found && !node_found_for_host) {
    // Add scanner node
}

return true
```

If we find a node, the second check will never be executed, which is why I've shortened this:
```
if (node_found) {
    // update hello_ok
    return true
}

if (!node_found_for_host) {
    // Add scanner node
}

return true
```

The check is the same as before, but I can change it back for clarity if you'd like.

Reply:
Doh... I missed the previous early return. Thank you for the explanation and for simplifying the change.